### ISSUE TITLE
Placement by cluster name requies managedcluster be labeled name=<cluster name>

### ISSUE THREAD

**Issue by hatfieldbrian** [open-cluster-management-io/multicloud-operators-subscription#16]
I wonder why a managedcluster needs to be labeled with its name, when its `metadata.name` could be used?

https://github.com/open-cluster-management-io/multicloud-operators-subscription/blob/3a52d8ba9c4595a002b48b34cd832c5b0690f5a9/pkg/utils/placementrule/placement.go#L52
If ocm requires a managedcluster be labeled with its name for placement, I wonder why ocm does not so label each managedcluster?

**Comment by mikeshng** [comment]
@xiangjingli do you know the history of this label requirement? 
As the author mentioned, if we have the name of the managedcluster it's probably good enough to just do a GET with the name.

**Comment by xiangjingli** [comment]
@mikeshng  The managedCluster CR has the name label by default in midstream.  So placementRule leverages the matchLabel functionality to get the clusters by cluster name.

Has the name label been removed from managedCluster CR in upstream? If so, we can switch to get the clusters by matching the managedCluster meta.name.

**Comment by mikeshng** [comment]
@xiangjingli the label doesn't seem to be there using just the upstream registration code. 

I think we can try doing label lookup first then if there are no results use a GET as a fallback? What do you think?

/assign mikeshng

**Comment by xiangjingli** [comment]
@mikeshng  yeah, it sounds good.  There are two ways

- get all managed clusters one time, return the managed clusters where the meta.name matches the cluster name list in placementRule spec.clusters.
- parse all the cluster names from the placementRule spec.clusters list. for each cluster name, get the managed cluster CR

It seems the first solution may get better performance. there is only one GET

**Comment by ivan-cai** [comment]
This problem still exists~

**Comment by mikeshng** [comment]
Hi @ivan-cai thank you for your feedback. We are currently in the process of adopting OCM's Placement API https://github.com/open-cluster-management-io/placement which is the native Placement API for OCM across the whole project. 
The current Subscription native PlacementRule API is designated for deprecation and will no longer be supported soon. Hence, we are unlikely to work on this item.

If the community is interested in this enhancement for the PlacementRule API, please feel free to submit a fix. Sorry for the lack of update until now.

**Comment by ivan-cai** [comment]
Hi @mikeshng，adopting Placement API is a good choice. But what time can it be done?

**Comment by mikeshng** [comment]
@ivan-cai we are targeting before end of this year.

**Comment by mikeshng** [comment]
We have implemented majority of the placement API integration. Follow steps in https://github.com/open-cluster-management-io/OCM/blob/main/solutions/deploy-a-helm-chart/READEME.md to give it a try. 

More docs and other misc changes to come later but for now the integration should work fine.

**Comment by nirs** [comment]
@mikeshng, We replaced usage of PlacementRule with Placement, do we still need to keep the [workaround](https://github.com/RamenDR/ramen/blob/975f2fbf6dceaccf59eff76dd4a65573cb6c4ded/test/addons/ocm-cluster/start#L98)  labeling the managed cluster?

**Comment by mikeshng** [comment]
> @mikeshng, We replaced usage of PlacementRule with Placement, do we still need to keep the [workaround](https://github.com/RamenDR/ramen/blob/975f2fbf6dceaccf59eff76dd4a65573cb6c4ded/test/addons/ocm-cluster/start#L98) labeling the managed cluster?

No, it shouldn't be needed but it's highly recommended you give it a try first.

Also, it might depend on your Placement spec. If your Placement spec contains the label criteria then it needs to be there.

@xiangjingli could you please confirm? Thanks.

**Comment by nirs** [comment]
> Also, it might depend on your Placement spec. If your Placement spec contains the label criteria then it needs to be there.

Thanks, will check.
